nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/CommandRunnerImpl.java

```diff
@@ -1642,12 +1643,22 @@ public void executeFromCheckpoint(JobManager.Checkpoint checkpoint, boolean reve
     private void publishCommandInvokedEvent(ExecutionContext invocation, Subject subject) {
         final CommandInvokedEvent event = new CommandInvokedEvent(
                 invocation.name(),
-                invocation.parameters(),
+                maskSecretParameters(invocation.parameters()),
                 subject);
         eventService.getCommandInvokedTopic()
                 .publish(event);
     }
 
+    private ParameterMap maskSecretParameters(ParameterMap parameters) {
```

Comment:
Could it be `static` and have unit test?

Reply:
OK. I will add this specific case into the CommandLoggerTest tests.